modules/core/src/v2/coins/abstractEthLikeCoin.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * @prettier
+ */
+import * as Bluebird from 'bluebird';
+import { CoinFamily, BaseCoin as StaticsBaseCoin } from '@bitgo/statics';
+
+import {
+  BaseCoin,
+  KeyPair,
+  ParsedTransaction,
+  ParseTransactionOptions,
+  SignedTransaction,
+  SignTransactionOptions,
+  VerifyAddressOptions,
+  VerifyTransactionOptions,
+  TransactionFee,
+  TransactionRecipient as Recipient,
+  TransactionPrebuild as BaseTransactionPrebuild,
+  TransactionExplanation,
+} from '../baseCoin';
+
+import { BitGo } from '../../bitgo';
+import { NodeCallback } from '../types';
+import { MethodNotImplementedError } from '../../errors';
+
+export interface EthSignTransactionOptions extends SignTransactionOptions {
+  txPrebuild: TransactionPrebuild;
+  prv: string;
+}
+
+export interface TxInfo {
+  recipients: Recipient[];
+  from: string;
+  txid: string;
+}
+
+export interface TransactionPrebuild extends BaseTransactionPrebuild {
+  txHex: string;
+  txInfo: TxInfo;
+  feeInfo: EthTransactionFee;
+  source: string;
+  dataToSign: string;
+}
+
+export interface EthTransactionFee {
+  fee: string;
+  gasLimit?: string;
+}
+
+export interface ExplainTransactionOptions {
+  txHex?: string;
+  halfSigned?: {
+    txHex: string;
+  };
+  feeInfo: TransactionFee;
+}
+
+export class AbstractEthLikeCoin extends BaseCoin {
```

Comment:
Why call it `Abstract` but not actually make it `abstract`?

Reply:
Similar to `abstractUTXOCoin`, it's meant to implement the basic shared logic that all ETH-like coins will need to implement anyways. I can change it to just `baseEthLikeCoin` or something if you like.

Reply:
No, it's fine. It's should probably be marked `abstract` just to prevent inadvertent instance creation, but it's fine to handle in a follow up PR.